code/datums/flock/flock.dm

```diff
@@ -447,24 +466,25 @@
 			// tile got killed, remove it
 			floors_no_longer_existing |= T
 			continue
-		// check adjacent tiles to see if we've been surrounded and can start generating, or if we're no longer surrounded and can't generate
-		// var/validNeighbors = 0
-		// var/list/neighbors = getNeighbors(T, cardinal)
-		// for(var/turf/simulated/floor/feather/F in neighbors)
-		// 	validNeighbors += 1
-		// if(validNeighbors < 4 && T.generating)
-		// 	T.off()
-		// else if(validNeighbors >= 4 && !T.generating)
-		// 	T.on()
 
 	if(floors_no_longer_existing.len > 0)
 		src.all_owned_tiles -= floors_no_longer_existing
 
+	for(var/datum/unlockable_flock_structure/ufs as anything in src.unlockableStructures)
+		ufs.process()
+
 /datum/flock/proc/convert_turf(var/turf/T, var/converterName)
 	src.unreserveTurf(converterName)
 	src.claimTurf(flock_convert_turf(T))
 	playsound(T, "sound/items/Deconstruct.ogg", 70, 1)
 
+///Unlock an achievement (string) if it isn't already unlocked
+/datum/flock/proc/achieve(var/str)
+	src.achievements |= str
+
+///Unlock an achievement (string) if it isn't already unlocked
+/datum/flock/proc/hasAchieved(var/str)
+	return (str in src.achievements)
```

Comment:
Probably should convert into macros, though probably should ask in #imcoder about that

Reply:
how so? achievements are going to be flock specific so if it was macro'd we'd just have to do hasAchieved(flock, "thing") instead of flock.hasAchieved("thing")

Reply:
I'm not to sure on this one really. This is probably fine actually.